crates/pixi_utils/src/conda_environment_file.rs

```diff
@@ -1,11 +1,53 @@
-use std::{io::BufRead, path::Path, str::FromStr};
-
 use itertools::Itertools;
-use miette::IntoDiagnostic;
+use miette::{Context, Diagnostic, IntoDiagnostic, NamedSource, SourceSpan};
+use pixi_config::Config;
 use rattler_conda_types::{MatchSpec, NamedChannelOrUrl, ParseStrictness::Lenient};
 use serde::Deserialize;
+use std::path::PathBuf;
+use std::{io::BufRead, path::Path, str::FromStr};
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+#[error("Failed to parse '{path}' as a conda environment file")]
+struct YamlParseError {
+    #[source]
+    source: serde_yaml::Error,
+    src: NamedSource<String>,
+    span: Option<SourceSpan>,
+    path: PathBuf,
```

Comment:
Neither `src` not `path` seem to be used anywhere

Reply:
They are, `src` is used in the `fn source_code` and `path` is used in the error msg itself (`#[error(".. '{path}'`).

Reply:
Indeed, my bad